scripts/dev_cli.sh: Script fails to build cloud-hypervisor on ARM

**Describe the bug**
Kata Containers CI is trying to unify the way we build and test Cloud Hypervisor, by leveraging the usage of the `scripts/dev_cli.sh`.
However, building Cloud Hypervisor on ARM is not currently working due to:
```
22:21:18 HEAD is now at 5343e09e scripts: dev_cli: Add `--features` option
22:21:18 20220223-0: Pulling from cloudhypervisor/dev
22:21:18 bbf2fb66fa6e: Already exists
22:21:19 adef39a94a86: Pulling fs layer
22:21:19 385b055b4506: Pulling fs layer
22:21:19 f8f99bc1f4a4: Pulling fs layer
22:21:19 155a08a3be7d: Pulling fs layer
22:21:19 a38769f13ce3: Pulling fs layer
22:21:19 d059cc9b73c9: Pulling fs layer
22:21:19 155a08a3be7d: Waiting
22:21:19 a38769f13ce3: Waiting
22:21:19 d059cc9b73c9: Waiting
22:21:19 385b055b4506: Download complete
22:21:19 155a08a3be7d: Verifying Checksum
22:21:19 155a08a3be7d: Download complete
22:21:19 adef39a94a86: Download complete
22:21:23 d059cc9b73c9: Download complete
22:21:23 f8f99bc1f4a4: Verifying Checksum
22:21:24 f8f99bc1f4a4: Download complete
22:21:24 a38769f13ce3: Verifying Checksum
22:21:25 a38769f13ce3: Download complete
22:21:25 adef39a94a86: Pull complete
22:21:26 385b055b4506: Pull complete
22:21:26 f8f99bc1f4a4: Pull complete
22:21:34 155a08a3be7d: Pull complete
22:21:35 a38769f13ce3: Pull complete
22:21:37 d059cc9b73c9: Pull complete
22:21:37 Digest: sha256:e7836c6ce9c498e6cfd6bc23603fc7eeda652d7c2af918ef04d3d2462e5f3d0f
22:21:37 Status: Downloaded newer image for cloudhypervisor/dev:20220223-0
22:21:37 docker.io/cloudhypervisor/dev:20220223-0
22:21:37     Updating git repository `[https://github.com/cloud-hypervisor/kvm-bindings`](https://github.com/cloud-hypervisor/kvm-bindings%60)
22:21:41     Updating git submodule `[https://github.com/rust-vmm/rust-vmm-ci.git`](https://github.com/rust-vmm/rust-vmm-ci.git%60)
22:21:42     Updating git repository `[https://github.com/rust-vmm/kvm-ioctls`](https://github.com/rust-vmm/kvm-ioctls%60)
22:21:42     Updating git submodule `[https://github.com/rust-vmm/rust-vmm-ci.git`](https://github.com/rust-vmm/rust-vmm-ci.git%60)
22:21:43     Updating git repository `[https://github.com/cloud-hypervisor/versionize_derive`](https://github.com/cloud-hypervisor/versionize_derive%60)
22:21:43     Updating git submodule `[https://github.com/rust-vmm/rust-vmm-ci.git`](https://github.com/rust-vmm/rust-vmm-ci.git%60)
22:21:43     Updating git repository `[https://github.com/rust-vmm/vm-virtio`](https://github.com/rust-vmm/vm-virtio%60)
22:21:44     Updating git submodule `[https://github.com/rust-vmm/rust-vmm-ci.git`](https://github.com/rust-vmm/rust-vmm-ci.git%60)
22:21:44     Updating crates.io index
22:21:45     Updating git repository `[https://github.com/rust-vmm/vm-fdt`](https://github.com/rust-vmm/vm-fdt%60)
22:22:15     Updating git submodule `[https://github.com/rust-vmm/rust-vmm-ci.git`](https://github.com/rust-vmm/rust-vmm-ci.git%60)
22:22:18     Updating git repository `[https://github.com/rust-vmm/mshv`](https://github.com/rust-vmm/mshv%60)
22:22:19     Updating git submodule `[https://github.com/rust-vmm/rust-vmm-ci.git`](https://github.com/rust-vmm/rust-vmm-ci.git%60)
22:22:19     Updating git repository `[https://github.com/rust-vmm/vfio`](https://github.com/rust-vmm/vfio%60)
22:22:20     Updating git submodule `[https://github.com/rust-vmm/rust-vmm-ci.git`](https://github.com/rust-vmm/rust-vmm-ci.git%60)
22:22:20     Updating git repository `[https://github.com/rust-vmm/vhost-user-backend`](https://github.com/rust-vmm/vhost-user-backend%60)
22:22:21     Updating git submodule `[https://github.com/rust-vmm/rust-vmm-ci.git`](https://github.com/rust-vmm/rust-vmm-ci.git%60)
22:22:21     Updating git repository `[https://github.com/firecracker-microvm/micro-http`](https://github.com/firecracker-microvm/micro-http%60)
22:22:21     Updating git submodule `[https://github.com/rust-vmm/rust-vmm-ci.git`](https://github.com/rust-vmm/rust-vmm-ci.git%60)
22:22:22  Downloading crates ...
22:22:22   Downloaded cfg-if v1.0.0
22:22:23   Downloaded bincode v1.3.3
22:22:23   Downloaded dirs v3.0.2
22:22:23   Downloaded versionize v0.1.6
22:22:23   Downloaded vhost-user-backend v0.1.0
22:22:23   Downloaded arc-swap v1.5.0
22:22:23   Downloaded remain v0.2.2
22:22:23   Downloaded epoll v4.3.1
22:22:23   Downloaded virtio-bindings v0.1.0
22:22:23   Downloaded pkg-config v0.3.24
22:22:23   Downloaded terminal_size v0.1.17
22:22:23   Downloaded thiserror-impl v1.0.30
22:22:23   Downloaded unicode-xid v0.2.2
22:22:23   Downloaded wait-timeout v0.2.0
22:22:23   Downloaded textwrap v0.14.2
22:22:23   Downloaded uuid v0.8.2
22:22:23   Downloaded managed v0.8.0
22:22:23   Downloaded openssl-sys v0.9.72
22:22:23   Downloaded lock_api v0.4.6
22:22:23   Downloaded serde_json v1.0.79
22:22:23   Downloaded regex v1.5.4
22:22:23   Downloaded anyhow v1.0.55
22:22:23   Downloaded clap v3.1.2
22:22:23   Downloaded aho-corasick v0.7.18
22:22:23   Downloaded syn v1.0.86
22:22:23   Downloaded cc v1.0.73
22:22:23   Downloaded byteorder v1.4.3
22:22:23   Downloaded semver v1.0.6
22:22:23   Downloaded bitflags v1.3.2
22:22:23   Downloaded autocfg v1.1.0
22:22:23   Downloaded vm-memory v0.7.0
22:22:23   Downloaded vmm-sys-util v0.9.0
22:22:23   Downloaded vhost v0.3.0
22:22:23   Downloaded env_logger v0.9.0
22:22:23   Downloaded vfio-bindings v0.3.1
22:22:23   Downloaded io-uring v0.5.2
22:22:23   Downloaded linux-loader v0.4.0
22:22:23   Downloaded seccompiler v0.2.0
22:22:23   Downloaded instant v0.1.12
22:22:23   Downloaded lazy_static v1.4.0
22:22:23   Downloaded log v0.4.14
22:22:23   Downloaded scopeguard v1.1.0
22:22:23   Downloaded quote v1.0.15
22:22:23   Downloaded rustc_version v0.4.0
22:22:23   Downloaded os_str_bytes v6.0.0
22:22:23   Downloaded signal-hook v0.3.13
22:22:23   Downloaded serde v1.0.136
22:22:23   Downloaded signal-hook-registry v1.4.0
22:22:23   Downloaded ryu v1.0.9
22:22:23   Downloaded humantime v2.1.0
22:22:23   Downloaded gdbstub_arch v0.2.2
22:22:23   Downloaded regex-syntax v0.6.25
22:22:23   Downloaded parking_lot v0.11.2
22:22:23   Downloaded dirs v4.0.0
22:22:23   Downloaded serde_derive v1.0.136
22:22:23   Downloaded libc v0.2.119
22:22:23   Downloaded num-traits v0.2.14
22:22:23   Downloaded libz-sys v1.1.3
22:22:23   Downloaded parking_lot_core v0.8.5
22:22:23   Downloaded gdbstub v0.6.1
22:22:23   Downloaded paste v1.0.6
22:22:23   Downloaded openssl-src v111.17.0+1.1.1m
22:22:23   Downloaded memchr v2.4.1
22:22:23   Downloaded itoa v1.0.1
22:22:23   Downloaded hashbrown v0.11.2
22:22:23   Downloaded indexmap v1.8.0
22:22:23   Downloaded getrandom v0.2.5
22:22:23   Downloaded dirs-sys v0.3.6
22:22:23   Downloaded libssh2-sys v0.2.23
22:22:23   Downloaded fdt v0.1.3
22:22:23   Downloaded crc32c v0.6.1
22:22:23   Downloaded crc64 v1.0.0
22:22:23   Downloaded ssh2 v0.9.3
22:22:23   Downloaded atty v0.2.14
22:22:23   Downloaded proc-macro2 v1.0.36
22:22:23   Downloaded thiserror v1.0.30
22:22:23   Downloaded smallvec v1.8.0
22:22:23   Downloaded termcolor v1.1.2
22:22:23   Downloaded strsim v0.10.0
22:22:23    Compiling libc v0.2.119
22:22:23    Compiling proc-macro2 v1.0.36
22:22:23    Compiling unicode-xid v0.2.2
22:22:23    Compiling syn v1.0.86
22:22:23    Compiling bitflags v1.3.2
22:22:23    Compiling serde v1.0.136
22:22:23    Compiling serde_derive v1.0.136
22:22:23    Compiling cfg-if v1.0.0
22:22:23    Compiling autocfg v1.1.0
22:22:23    Compiling log v0.4.14
22:22:23    Compiling arc-swap v1.5.0
22:22:23    Compiling memchr v2.4.1
22:22:23    Compiling serde_json v1.0.79
22:22:23    Compiling crc64 v1.0.0
22:22:23    Compiling byteorder v1.4.3
22:22:23    Compiling ryu v1.0.9
22:22:23    Compiling itoa v1.0.1
22:22:23    Compiling anyhow v1.0.55
22:22:23    Compiling semver v1.0.6
22:22:23    Compiling cc v1.0.73
22:22:23    Compiling kvm-ioctls v0.11.0 (https://github.com/rust-vmm/kvm-ioctls?branch=main#1e03e29c)
22:22:23    Compiling virtio-bindings v0.1.0
22:22:23    Compiling pkg-config v0.3.24
22:22:23    Compiling hashbrown v0.11.2
22:22:23    Compiling lazy_static v1.4.0
22:22:23    Compiling termcolor v1.1.2
22:22:23    Compiling io-uring v0.5.2
22:22:23    Compiling strsim v0.10.0
22:22:23    Compiling vm-fdt v0.2.0 (https://github.com/rust-vmm/vm-fdt?branch=main#dccd69c9)
22:22:23    Compiling fdt v0.1.3
22:22:23    Compiling option_parser v0.1.0 (/cloud-hypervisor/option_parser)
22:22:23    Compiling parking_lot_core v0.8.5
22:22:23    Compiling regex-syntax v0.6.25
22:22:23    Compiling humantime v2.1.0
22:22:23    Compiling signal-hook v0.3.13
22:22:23    Compiling scopeguard v1.1.0
22:22:23    Compiling paste v1.0.6
22:22:23    Compiling managed v0.8.0
22:22:23    Compiling smallvec v1.8.0
22:22:23    Compiling instant v0.1.12
22:22:23    Compiling lock_api v0.4.6
22:22:23    Compiling indexmap v1.8.0
22:22:24    Compiling num-traits v0.2.14
22:22:24    Compiling os_str_bytes v6.0.0
22:22:24    Compiling aho-corasick v0.7.18
22:22:24    Compiling openssl-src v111.17.0+1.1.1m
22:22:24    Compiling rustc_version v0.4.0
22:22:24    Compiling libz-sys v1.1.3
22:22:24    Compiling openssl-sys v0.9.72
22:22:24    Compiling libssh2-sys v0.2.23
22:22:24    Compiling quote v1.0.15
22:22:24    Compiling crc32c v0.6.1
22:22:25    Compiling vm-memory v0.7.0
22:22:25    Compiling epoll v4.3.1
22:22:25    Compiling getrandom v0.2.5
22:22:25    Compiling terminal_size v0.1.17
22:22:25    Compiling atty v0.2.14
22:22:25    Compiling signal-hook-registry v1.4.0
22:22:25    Compiling dirs-sys v0.3.6
22:22:25    Compiling seccompiler v0.2.0
22:22:25    Compiling wait-timeout v0.2.0
22:22:25    Compiling textwrap v0.14.2
22:22:25 error: failed to run custom build command for `libz-sys v1.1.3`
22:22:25 
22:22:25 Caused by:
22:22:25   process didn't exit successfully: `/cloud-hypervisor/build/cargo_target/release/build/libz-sys-486ad7227acbea5e/build-script-build` (exit status: 1)
22:22:25   --- stdout
22:22:25   cargo:rerun-if-env-changed=LIBZ_SYS_STATIC
22:22:25   cargo:rerun-if-changed=build.rs
22:22:25   cargo:rerun-if-env-changed=ZLIB_NO_PKG_CONFIG
22:22:25   cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_aarch64-unknown-linux-musl
22:22:25   cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_aarch64_unknown_linux_musl
22:22:25   cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_ALLOW_CROSS
22:22:25   cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS
22:22:25   cargo:rerun-if-env-changed=PKG_CONFIG_aarch64-unknown-linux-musl
22:22:25   cargo:rerun-if-env-changed=PKG_CONFIG_aarch64_unknown_linux_musl
22:22:25   cargo:rerun-if-env-changed=TARGET_PKG_CONFIG
22:22:25   cargo:rerun-if-env-changed=PKG_CONFIG
22:22:25   cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64-unknown-linux-musl
22:22:25   cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64_unknown_linux_musl
22:22:25   cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_SYSROOT_DIR
22:22:25   cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
22:22:25   TARGET = Some("aarch64-unknown-linux-musl")
22:22:25   OPT_LEVEL = Some("3")
22:22:25   HOST = Some("aarch64-unknown-linux-gnu")
22:22:25   CC_aarch64-unknown-linux-musl = None
22:22:25   CC_aarch64_unknown_linux_musl = None
22:22:25   TARGET_CC = None
22:22:25   CC = None
22:22:25   CROSS_COMPILE = None
22:22:25   CFLAGS_aarch64-unknown-linux-musl = None
22:22:25   CFLAGS_aarch64_unknown_linux_musl = None
22:22:25   TARGET_CFLAGS = None
22:22:25   CFLAGS = None
22:22:25   CRATE_CC_NO_DEFAULTS = None
22:22:25   DEBUG = Some("false")
22:22:25   CARGO_CFG_TARGET_FEATURE = None
22:22:25   running: "aarch64-linux-musl-gcc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-I" "src/zlib" "-fvisibility=hidden" "-DSTDC" "-D_LARGEFILE64_SOURCE" "-D_POSIX_SOURCE" "-o" "/cloud-hypervisor/build/cargo_target/aarch64-unknown-linux-musl/release/build/libz-sys-cdd4206b88a8ad1e/out/lib/src/zlib/adler32.o" "-c" "src/zlib/adler32.c"
22:22:25 
22:22:25   --- stderr
22:22:25 
22:22:25 
22:22:25   error occurred: Failed to find tool. Is `aarch64-linux-musl-gcc` installed?
22:22:25 
22:22:25 
22:22:25 warning: build failed, waiting for other jobs to finish...
22:22:25 error: build failed
22:22:31 make[1]: *** [tools/packaging/kata-deploy/local-build/Makefile:30: cloud-hypervisor-tarball-build] Error 101
22:22:31 make[1]: Leaving directory '/home/jenkins/workspace/kata-containers-2.0-ubuntu-ARM-PR/go/src/github.com/kata-containers/kata-containers'
22:22:31 make: *** [tools/packaging/kata-deploy/local-build/Makefile:33: cloud-hypervisor-tarball] Error 2
22:22:31 Failed to run 'override-build': Exit code was 2.
```

It's important to note that his is happening inside the following container: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/resources/Dockerfile

**To Reproduce**
Call `./scripts/dev_cli.sh build --release --libc musl` from an ARM machine.

**Version**

Kata Containers is currently using the following commit from Cloud Hypervisor:
5343e09e7b8dbd5dd8ac0d90a3ad52037490dd86

[...]

--- a/scripts/dev_cli.sh
+++ b/scripts/dev_cli.sh
@@ -275,8 +275,10 @@ cmd_build() {
     [ "$(uname -m)" = "aarch64" ] && cargo_args+=(--features "$hypervisor")
 
     rustflags=""
+    target_cc=""
     if [ "$(uname -m)" = "aarch64" ] && [ "$libc" = "musl" ]; then
         rustflags="-C link-arg=-lgcc -C link_arg=-specs -C link_arg=/usr/lib/aarch64-linux-musl/musl-gcc.specs"
+        target_cc="musl-gcc"
     fi
 
     $DOCKER_RUNTIME run \
@@ -286,6 +288,7 @@ cmd_build() {
         --volume $exported_device \
         --volume "$CLH_ROOT_DIR:$CTR_CLH_ROOT_DIR" $exported_volumes \
         --env RUSTFLAGS="$rustflags" \
+        --env TARGET_CC="$target_cc" \
         "$CTR_IMAGE" \
         cargo build --all $features_build \
         --target-dir "$CTR_CLH_CARGO_TARGET" \

